base/compiler/ssair/passes.jl

```diff
@@ -805,6 +804,10 @@ end
         rarg = argdef.args[1]
         isa(rarg, SSAValue) || return nothing
         argdef = compact[rarg][:inst]
+    else
+        isa(arg, DataType) || return nothing
+        isType(arg) || return nothing
```

Comment:
```suggestion
        isType(arg) || return nothing
```
`isType(x)` includes `isa(x, DataType)`.

Reply:
Will do in a follow up